server/test/repository/ExportServiceRepositoryTest.java

```diff
@@ -0,0 +1,137 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Locale;
+import java.util.NoSuchElementException;
+import models.LifecycleStage;
+import models.Question;
+import org.junit.Before;
+import org.junit.Test;
+import services.LocalizedStrings;
+import services.question.QuestionOption;
+import services.question.QuestionService;
+import services.question.types.MultiOptionQuestionDefinition;
+import services.question.types.QuestionDefinition;
+import services.question.types.QuestionDefinitionBuilder;
+import services.question.types.QuestionDefinitionConfig;
+
+public class ExportServiceRepositoryTest extends ResetPostgres {
+  private ExportServiceRepository repo;
+  // private Applicant applicant;
+  // private Program program;
+  private QuestionService questionService;
+  private VersionRepository versionRepository;
+
+  @Before
+  public void setUp() {
+    repo = instanceOf(ExportServiceRepository.class);
+    questionService = instanceOf(QuestionService.class);
+    versionRepository = instanceOf(VersionRepository.class);
+  }
+
+  @Test
+  public void getMultiSelectedHeaders_NoPreviousQuestionVersion() {
+    QuestionDefinition questionDefinition =
+        testQuestionBank.applicantKitchenTools().getQuestionDefinition();
+    ImmutableList<String> multiSelectHeaders = repo.getMultiSelectedHeaders(questionDefinition);
+    assertThat(multiSelectHeaders.size()).isEqualTo(3);
+    checkList(multiSelectHeaders, 0, "toaster");
+    checkList(multiSelectHeaders, 1, "pepper_grinder");
+    checkList(multiSelectHeaders, 2, "garlic_press");
+  }
+
+  @Test
+  public void getMultiSelectedHeaders_DraftVersionExcludedButPublishedVersionIncluded()
+      throws Exception {
+    Question questionWeather =
+        createMultiSelectQuestion("weather", "fall", "spring", "summer", LifecycleStage.ACTIVE);
+    MultiOptionQuestionDefinition multiOptionQuestionDefinition =
+        (MultiOptionQuestionDefinition) questionWeather.getQuestionDefinition();
+    QuestionOption newOption =
+        QuestionOption.create(4L, 4L, "winter", LocalizedStrings.of(Locale.US, "winter"));
+    ImmutableList<QuestionOption> currentOptions = multiOptionQuestionDefinition.getOptions();
+    ImmutableList<QuestionOption> newOptionList =
+        ImmutableList.<QuestionOption>builder().addAll(currentOptions).add(newOption).build();
+    QuestionDefinition toUpdate =
+        new QuestionDefinitionBuilder(questionWeather.getQuestionDefinition())
+            .setQuestionOptions(newOptionList)
+            .build();
+    questionService.update(toUpdate);
+
+    ImmutableList<String> multiSelectHeaders =
+        repo.getMultiSelectedHeaders(questionWeather.getQuestionDefinition());
+    // not draft versions are not part of the header list
+    assertThat(multiSelectHeaders.size()).isEqualTo(3);
+    versionRepository.publishNewSynchronizedVersion();
```

Comment:
**style**: adding aline break before this line will help separate the two "act and assert" blocks.

(Arranging test logic into "arrange", "act", and "assert" code blocks makes them more readable.)